agents/fts_stage/fts_agent.py

```diff
@@ -0,0 +1,304 @@
+        # so does not require the lock
+        self.initialized = True
+        if self.auto_acq:
+            self.agent.start('acq')
+        return True, 'Stage Initialized.'
+
+    def home_task(self, session, params=None):
+        """ Home the stage to its negative limit
+        """
+
+        with self.lock.acquire_timeout(timeout=3, job='home') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start home because lock held by" \
+                                "{self.lock.job}")
+                return False, "Could not get lock"
+            self.stage.home()
+
+        return True, "Homing Complete"
+    
+    def move_to(self, session, params):
+        """Move to absolute position relative to stage center (in mm)
+
+        params: {'position':float between -74.8 and 74.8}
+        """
+    
+        with self.lock.acquire_timeout(timeout=3, job='move') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start move because lock held by" \
+                                "{self.lock.job}")
+                return False, "Could not get lock"
+            self.stage.move_to( params.get('position') )
+
+        return True, "Move Complete"
+    
+    def start_acq(self, session, params=None):
+        """
+        params:
+            dict: {'sampling_frequency': float, sampling rate in Hz}
+        """
+        if params is None:
+            params = {}
+
+
+        f_sample = params.get('sampling_frequency', self.sampling_frequency)
+        pm = Pacemaker(f_sample, quantize=True)
+
+        if not self.initialized or self.stage is None:
+            raise Exception("Connection to Stages not initialized")
+
+        with self.lock.acquire_timeout(timeout=0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already" \
+                            "running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            self.log.info(f"Starting Data Acquisition for FTS Mirror at" \
+                            "{f_sample} Hz")
+            session.set_status('running')
+            self.take_data = True
+            last_release = time.time()
+
+            while self.take_data:
+                if time.time()-last_release > 1.:
+                    if not self.lock.release_and_acquire(timeout=20):
+                        self.log.warn(f"Could not re-acquire lock now held by" \
+                                      "{self.lock.job}.")
+                        return False, "could not re-acquire lock"
+                    last_release = time.time()
+                pm.sleep()
+
+                data = {
+                    'timestamp':time.time(),
+                    'block_name':'position',
+                    'data':{}}
+                success, pos = self.stage.get_position()
+                
+                data['data']['pos'] = pos
```

Comment:
If `get_position()` isn't successful this is going to put a `None` into the data array, which will cause a `TypeError` when published. We should prevent that from happening. Perhaps check the success state before publishing?